Week 1/Extra Practice Problems (Week 1)/0496: Next Greater Element I/dominic.cpp

```diff
@@ -0,0 +1,21 @@
+class Solution {
+public:
+    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
+        unordered_map<int, int> m;
+        stack<int> s;
+        for (auto &x : nums2) {
+            while (!s.empty() && x > s.top()) {
+                m[s.top()] = x;
+                s.pop();
+            }
+            s.push(x);
+        }
+
+        vector<int> res;
```

Comment:
we can make res the same size of nums1 to avoid resizing

Reply:
Updated with no resizing!